server_nginx.html

```diff
@@ -15,36 +15,40 @@
 # Wide-open CORS config for nginx
 #
 location / {
+     # the dance you have to do because if's can't be nested in nginx
+     if ($http_origin) {
+        set $cors "${cors}O"
+     }
      if ($request_method = 'OPTIONS') {
-        add_header 'Access-Control-Allow-Origin' '*';
-        #
-        # Om nom nom cookies
-        #
-        add_header 'Access-Control-Allow-Credentials' 'true';
-        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
-        #
-        # Custom headers and headers various browsers *should* be OK with but aren't
-        #
-        add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
-        #
-        # Tell client that this pre-flight info is valid for 20 days
-        #
-        add_header 'Access-Control-Max-Age' 1728000;
-        add_header 'Content-Type' 'text/plain charset=UTF-8';
-        add_header 'Content-Length' 0;
-        return 204;
+        set $cors "${cors}P"
      }
      if ($request_method = 'POST') {
-        add_header 'Access-Control-Allow-Origin' '*';
-        add_header 'Access-Control-Allow-Credentials' 'true';
-        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
-        add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
+        $cors = "${cors}S"
      }
      if ($request_method = 'GET') {
-        add_header 'Access-Control-Allow-Origin' '*';
-        add_header 'Access-Control-Allow-Credentials' 'true';
-        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+        $cors = "${cors}G"
+     }
+
+     # ORIGIN + OPTION request
+     if ($cors = OP) {
+        add_header 'Access-Control-Allow-Origin' $http_origin;
+        # these are only needed on OPTIONS
+        add_header 'Access-Control-Allow-Credentials' 'true'; # techically only needed if doing credentials
+
+        # be sure to add any other headers needed eg. X-APIKEY, Cache-Control...
         add_header 'Access-Control-Allow-Headers' 'DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
```

Comment:
i just copied from https://github.com/monsur/enable-cors.org/blob/gh-pages/server_nginx.html#L24 so yeah you're right but i didn't fix it in the latest commit. 

--timball